commit 3735ef3b3804aaa84749e403667bac3674970f02
Author: 12101111 <w12101111@gmail.com>
Date:   Tue Mar 30 10:18:05 2021 +0800

    Revert "cert: Use the new autodestruct syntax"
    
    This reverts commit 9b3d730847e1a8d182eba4a50b6d9b51fe137617.

diff --git a/ell/cert.c b/ell/cert.c
index 14631b2..21e54c4 100644
--- a/ell/cert.c
+++ b/ell/cert.c
@@ -435,6 +434,11 @@ static struct l_key *cert_try_link(struct l_cert *cert, struct l_keyring *ring)
 	return NULL;
 }
 
+static void cert_keyring_cleanup(struct l_keyring **p)
+{
+	l_keyring_free(*p);
+}
+
 #define RETURN_ERROR(msg, args...)	\
 	do {	\
 		if (error) {	\
@@ -449,7 +453,8 @@ LIB_EXPORT bool l_certchain_verify(struct l_certchain *chain,
 					const char **error)
 {
 	struct l_keyring *ca_ring = NULL;
-	_auto_(l_keyring_free) struct l_keyring *verify_ring = NULL;
+	L_AUTO_CLEANUP_VAR(struct l_keyring *, verify_ring,
+				cert_keyring_cleanup) = NULL;
 	struct l_cert *cert;
 	struct l_key *prev_key = NULL;
 	int verified = 0;
